snuba/web/query.py

```diff
@@ -195,48 +224,87 @@ def raw_query(
     return result
 
 
-def log_query_and_update_stats(
+def update_query_list_and_stats(
     request: Request,
     sql: str,
     timer: Timer,
     stats: MutableMapping[str, Any],
-    status: str,
+    query_list: MutableSequence[ClickhouseQueryMetadata],
     query_settings: Mapping[str, Any],
+    trace_id: str,
+    status: str,
 ) -> MutableMapping:
     """
     If query logging is enabled then logs details about the query and its status, as
     well as timing information.
     Also updates stats with any relevant information and returns the updated dict.
     """
-
     stats.update(query_settings)
+
+    query_list.append(
+        ClickhouseQueryMetadata(
+            request=request.body,
+            sql=sql,
+            timer=timer,
+            stats=stats,
+            status=status,
+            trace_id=trace_id,
+        )
+    )
+
+    return stats
+
+
+def record_query(request: Request, timer: Timer, query_list: MutableSequence[ClickhouseQueryMetadata]) -> None:
     if settings.RECORD_QUERIES:
+        last_query = query_list[-1]
         # send to redis
         state.record_query(
             {
                 "request": request.body,
-                "sql": sql,
-                "timing": timer,
-                "stats": stats,
-                "status": status,
+                "referrer": http_request.referrer,
+                "timing": last_query.timer.for_json(),
+                "status": last_query.status,
+                "query_list": [q.to_dict() for q in query_list],
```

Comment:
This is going to break the snuba dashboard I guess, since it changes the format of what we store in Redis. 
https://github.com/getsentry/snuba/blob/master/snuba/web/views.py#L161

Well, technically, it is going to break it more than it is already since that dashboard is broken.

Reply:
I forgot about that, fixed now.